salt/modules/aptpkg.py

```diff
@@ -384,7 +390,14 @@ def refresh_db(cache_valid_time=0, failhard=False, **kwargs):
         except OSError as exp:
             log.warning("could not stat cache directory due to: %s", exp)
 
-    call = _call_apt(["apt-get", "-q", "update"], scope=False)
+    call = _call_apt(
+        ["apt-get", "-q", "update"],
+        scope=False,
+        timeout=kwargs.get("timeout", __opts__.get("aptpkg_refresh_db_timeout", 30)),
+    )
+    if "Timed out" in call["stdout"]:
```

Comment:
The retcode when a timeout occurs is 1 so maybe check that first.

cmdmod.py 831-838:
 
```
except TimedProcTimeoutError as exc:
                    ret["stdout"] = str(exc)
                    ret["stderr"] = ""
                    ret["retcode"] = None
                    ret["pid"] = proc.process.pid
                    # ok return code for timeouts?
                    ret["retcode"] = 1
```